lib/plugins/aws/package/compile/imports/imports.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const traverse = require('traverse');
+
+class AwsCompileImports {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.compileImports = this.compileImports.bind(this);
+
+    this.hooks = {
+      'package:compileImports': this.compileImports,
+    };
+  }
+
+  compileImports() {
+    const that = this; // save this because of the upcoming scope switch
+    traverse(this.serverless.service).forEach(function (propertyParam) {
+      if (typeof propertyParam === 'string' && propertyParam.match(/imports:.+/)) {
+        const removedPrefix = propertyParam.split(':')[1];
+        const parts = removedPrefix.split('.');
+        const stack = parts[0];
```

Comment:
This is the part the user defines in the `imports` config section.

So he needs to explicitly append the stage to the stack name:

```yml
imports:
  my-service-dev:
    - foo
    - bar
```